unumpy/_multimethods.py

```diff
@@ -824,25 +804,442 @@ def nancumsum(a, axis=None, dtype=None, out=None):
     return (a, mark_dtype(dtype), mark_non_coercible(out))
 
 
-@create_numpy(_reduce_argreplacer)
+@create_numpy(_first2argreplacer)
 @all_of_type(ndarray)
-def std(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+def percentile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def nanpercentile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def quantile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def nanquantile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+def _ureduce(a, axis):
+    nd = ndim(a)
+
+    if axis is None:
+        a = ravel(a)
+        dims = (1,) * nd
+    else:
+        if not isinstance(axis, collections.abc.Sequence):
+            axis = (axis,)
+
+        axis = _normalize_axis(nd, axis)
+        unselected_axis = tuple(set(range(nd)) - set(axis))
+
+        dims = list(a.shape)
+        for ax in axis:
+            dims[ax] = 1
+        dims = tuple(dims)
+
+        a = transpose(a, unselected_axis + axis)
```

Comment:
Why do you transpose and call functions with `axis=-1` when you could just call the basic functions with `axis=axis`?

Reply:
I suggested that during the meeting to make the implementation simpler.

Reply:
If I'm not mistaken it was necessary for `median`'s default which led me to using it on the other reduce methods as well. But now that you point that out it's possible that the only method that needs it is `median`.

Reply:
I'll remove `_reduce` from the methods where it's not needed.